Src/Base/AMReX_ParallelDescriptor.H

```diff
@@ -1005,6 +1013,79 @@ ParallelDescriptor::Gatherv (const T* send, Long sc,
     BL_COMM_PROFILE(BLProfiler::Gatherv, std::accumulate(rc.begin(),rc.end(),0L)*sizeof(T), root, BLProfiler::NoTag());
 }
 
+template <class T>
+void
+ParallelDescriptor::GatherLayoutDataToVector (const LayoutData<T>& sendbuf,
+                                              Vector<T>& recvbuf, int root)
+{
+    BL_PROFILE_T_S("ParallelDescriptor::GatherLayoutData(Ti)", T);
+    
+#ifdef BL_USE_MPI
+    // Gather prelims
+    Vector<int> index_to_send = sendbuf.IndexArray();
+    Vector<T> T_to_send;
+    T_to_send.reserve(sendbuf.local_size());
+    
+    for (auto i : sendbuf.IndexArray())
+    {
+        T_to_send.push_back(sendbuf[i]);
+    }
```

Comment:
I switched back to the way as written from original revision you suggested:

> for (MFIter mfi(sendbuf); mfi.isValid(); ++mfi)
> {   
>     T_to_send.push_back(sendbuf[mfi]);
> }

after moving the function to AMReX_ParallelDescriptor.H.  Is this way OK, or do you recommend that we make the necessary changes to allow using the MFIter loop?